src/cockpit/389-console/src/lib/security/certificateManagement.jsx

```diff
@@ -239,24 +363,211 @@ export class CertificateManagement extends React.Component {
                         msg = errMsg.desc + " - " + errMsg.info;
                     }
                     this.setState({
+                        showExportModal: false,
+                        exportNickname: '',
                         modalSpinning: false,
-                        loading: false,
                     });
                     this.props.addNotification(
                         "error",
-                        `Error adding certificate - ${msg}`
+                        `Error exporting certificate - ${msg}`
                     );
                 });
     }
 
+    deleteTmpCert (filename) {
+        const cmd = [
+            'rm',
+            filename
+        ];
+
+        log_cmd("deleteTmpCert", "deleting tmp cert file", cmd);
+        cockpit
+                .spawn(cmd, { superuser: true, err: "message" })
+                .fail((err) => {
+                    this.props.addNotification(
+                        "warning",
+                        `Error deleting tmp certificate - ${err}`
+                    );
+                });
+    }
+
+    getCertFiles () {
+        const cmd = [
+            //'/bin/sh', '-c',
+            'find',
+            this.props.certDir,
+            '-type',
+            'f',
+            '-name',
+            '*.pem',
+            '-o',
+            '-name',
+            '*.crt',
+        ];
+        log_cmd("getCertFiles", "creating tmp cert file for importing", cmd);
+        cockpit
+                .spawn(cmd, { superuser: true, err: "message" })
+                .done((certs_raw) => {
+                    const certs = certs_raw.split(/\r?\n/);
+                    let certNames = [];
+                    for (let i = 0, count = 0; i < certs.length; i++) {
+                        if (certs[i] !== "") {
+                            certNames[count] = certs[i].replace(this.props.certDir + "/", "");
+                            count++;
+                        }
+                    }
+                    certNames.sort();
+
+                    let selectCertName = "";
+                    if (certNames.length > 0) {
+                        selectCertName = certNames[0];
+                    }
+                    this.setState({
+                        availCertNames: certNames,
+                        selectCertName,
+                    })
+                })
+                .fail(err => {
+                    console.log("Failed to get cert file names: ", err);
+                });
+    }
+
+    addCert (isCACert) {
+        this.setState({
+            modalSpinning: true,
+            loading: true,
+        });
+
+        let certType = "certificate";
+        if (isCACert) {
+            certType = "ca-certificate";
```

Comment:
Okay, sorry, it's once again me with another error I found...

When I export in binary->delete old->add the exported = it fails for me with the next error:
`Danger alert:
Error adding certificate - 'utf-8' codec can't decode byte 0x82 in position 1: invalid start byte`

The video:
[Screencast from 02-06-2023 03:12:04 PM.webm](https://user-images.githubusercontent.com/5634892/217108760-24065802-3b02-4fe6-8b9c-33d2d9725722.webm)

The same is true for Server Cert.

Reply:
Yeah perhaps certutil is not what you use for .crt files?  pk12util?   Maybe different flags are needed?  Not sure I'll look into it tomorrow.

Reply:
Ok the problem is that lib389 expects that we only work with PEM files.   A minor tweak to lib389 fixed this.  Please review one last time (hopefully)  :-)

Reply:
For some reason, it still doesn't work with CA certs... (Server certs are fine now!)

Reply:
Looking at the code, may be it's because `add_ca_cert_bundle` works with PEM (hence, failing with binary import):
https://github.com/389ds/389-ds-base/blob/b8c90a5a757099f466326c1ffc68cb89c74c0b14/src/lib389/lib389/nss_ssl.py#L1233-L1240